cpp/src/neighbors/detail/vpq_dataset.cuh

```diff
@@ -140,7 +140,7 @@ auto train_vq(const raft::resources& res, const vpq_params& params, const Datase
   using kmeans_in_type = typename DatasetT::value_type;
   cuvs::cluster::kmeans::balanced_params kmeans_params;
   kmeans_params.n_iters = params.kmeans_n_iters;
-  kmeans_params.metric  = cuvs::distance::DistanceType::L2Expanded;
+  kmeans_params.metric  = params.metric;
```

Comment:
Yes I have verified that the inner product kernel is being used.

Reply:
How about the same change on lines 170 and 230?

Reply:
We always train the product quantizer using L2 (we do the same with IVF-PQ). So that every sub-vector can be represented as <VQ centroid + PQ centroid>

Reply:
Ok we needed the metric to be inner product for VQ prediction. I have made the update. Marking this as resolved.